### PR TITLE
Rewrite some va.gov urls to staging urls

### PR DESCRIPTION
**Pull request by jbalboni**
## Description

This searches for particular domain strings and replaces them with others, in non-production environments. We're doing this to support linking to va.gov staging domains for some links while on our own staging and development sites, so we can see the injected headers on those pages before we go live.

This is set up to work in the static pages and also in the mega menu and footer React components. I haven't created anything to handle this in React apps because we should have fairly few links to these pages in those apps, and because I don't yet know the exact domains we're going to be replacing. So I'm punting on that for now.

## Testing done

Tested locally that the links we should update are updated and site still works

## Screenshots
screen shot 2018-09-26 at 4 07 55 pm

## Acceptance criteria
- [x] Links are replaced in non-prod environments
- [x] Links aren't replaced in prod environments

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/utilities/environment/stagingDomains.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+// This list also exists in script/options.js
+const domainReplacements = [
+  { from: 'www\\.cem\\.va\\.gov', to: 'staging.cem.va.gov' }
+];
```
Comment (ncksllvn) [inline src/platform/utilities/environment/stagingDomains.js]: I think this data would have to be passed through the `create-build-settings.js` middleware and referenced through the window. We probably need some sort of module to make lookups to that data more centralized, versus global access.

Comment (ncksllvn) [reply src/platform/utilities/environment/stagingDomains.js]: I'm going to explore some other ideas, but at the moment, that might be the method of passing data.

**Comment by jbalboni** [comment]
There's going to be a later ticket to add more staging subdomains, once we get those back.

File: script/rewrite-va-domains.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+          file.contents = new Buffer(contents);
+        });
+    }
+
+    done();
+  };
+}
+
+module.exports = createRedirects;
```
Comment (ncksllvn) [inline script/rewrite-va-domains.js]: I feel like we're getting pretty good at writing redirect middleware at this point 😆

File: src/platform/utilities/tests/environment/stagingDomains.unit.spec.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      expect(replaceDomainsInData(data)).to.deep.equal({
+        test: 'www.va.gov/testing',
+        other: {
+          href: `${currentEnv}.va.gov`,
+        }
+      });
+    });
+  });
+});
+
```
Comment (ncksllvn) [inline src/platform/utilities/tests/environment/stagingDomains.unit.spec.js]: 🔥

**Review by ncksllvn — APPROVED** [review]
LGTM!

**Pull request merged**